sdk/core/Azure.Core/src/Pipeline/Internal/ResponseBodyPolicy.cs

```diff
@@ -37,7 +37,8 @@ private async ValueTask ProcessSyncOrAsync(HttpMessage message, ReadOnlyMemory<H
             // We either use the default that the policy was constructed with at
             // pipeline-creation time, or we get an override value from the message that
             // we use for the duration of this invocation only.
-            TimeSpan invocationNetworkTimeout = message.NetworkTimeout ?? _networkTimeout;
+            message.NetworkTimeout ??= _networkTimeout;
```

Comment:
I'm wondering if setting this on the message could cause confusion for protocol method scenarios in which we are mutating the message provided by the developer in a non-obvious way.